src/server/mod.rs

```diff
@@ -948,4 +961,35 @@ where
             },
         }
     }
+
+    fn handle_control_channel_error(error: tokio::timer::timeout::Error<FTPError>, with_metrics: bool) -> Reply {
+        return if error.is_timer() {
+            // Timer errors can either be due to the timer being shutdown or being at_capacity.
+            // In both cases we can simply tell the client that the service is unavailable.
+            Reply::new(ReplyCode::ServiceNotAvailable, "Service not available, please try again later")
+        } else if error.is_inner() {
+            match error.into_inner() {
+                Some(ftp_error) => {
+                    if with_metrics {
+                        metrics::add_error_metric(ftp_error.kind());
```

Comment:
Do we not count the "unknown" errors? (ie have a line like this in the other branches)

Reply:
I looked at the metrics that we currently have and wasn't really sure how to fit these errors into those. It would be nice to add specific metrics, especially in the cases where we can (eg. in shutdown and at capacity). I haven't yet had the time for that and most probably I won't be able to work on this again until Friday next week, so I wanted to push what I had so far.

Reply:
Fair enough

Reply:
Metrics have been added.